src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[][] salaryByName = new String[names.length][2];
+        for (int i = 0; i < names.length; i++) {
+            salaryByName[i][0] = names[i];
+            salaryByName[i][1] = "0";
+        }
+        String[][] splitedData = splitData(data);
+        for (int i = 0; i < splitedData.length; i++) {
+            if (dayCheck(splitedData[i][0], dateFrom, dateTo)) {
+                for (int j = 0; j < names.length; j++) {
+                    if (splitedData[i][1].equals(salaryByName[j][0])) {
+                        try {
+                            salaryByName[j][1] =
+                                    String.valueOf((Integer.parseInt(salaryByName[j][1])
+                                    + (Integer.parseInt(splitedData[i][2])
+                                    * Integer.parseInt(splitedData[i][3]))));
+                        } catch (NumberFormatException ex) {
+                            System.out.println("Error parsing string to number: "
+                                    + ex.getMessage());
+                        }
+                    }
+                }
+            }
```

Comment:
Don't create redundant loops. Two loops are enough to solve this task. Try to optimize the code to use only two loops.